src/symfc_alm/ridge.py

```diff
@@ -238,19 +261,43 @@ def _calc_error(self, alpha: float):
             The mean squared error calculated by LOOCV.
 
         """
+        if alpha is not None:
+            self._RTR_inv = self._calc_RTR_inv(alpha)
         b_pred = self._predict(self._A)
-        D_inv_sqrt = np.diag(1 / np.sqrt(self.__D + alpha))
-        AUD_inv_sqrt = np.dot(self._A, np.dot(self.__U, D_inv_sqrt))
-
-        n = AUD_inv_sqrt.shape[0]  # number of rows
-        error = 0.0
-        for j in range(n):
-            beta_j = np.sum(np.dot(AUD_inv_sqrt[j, :], AUD_inv_sqrt[j, :]))  # H[i, i]
-            error += ((self._b[j] - b_pred[j]) / (1 - beta_j)) ** 2
+        H = self._R @ self._RTR_inv @ self._R.T
+        n = H.shape[0]
+        error = 0
+        for i in range(n):
+            error += ((self._b[i] - b_pred[i]) / (1 - H[i, i])) ** 2
         error /= n
 
         return error
 
+    def _calc_RTR_inv(self, alpha: float):
+        r"""Return the inverse of a specific matrix.
+
+        The inverse is calculated for the following equation:
+
+        (R^{T}R + \alpha I)^{-1}
+
+        where R is R = U @ Σ, and U, Σ are obtained from the
```

Comment:
Please avoid using Greek letter character.

Reply:
Thank you very much. I fixed it.